store/HookStore.ts

```diff
@@ -0,0 +1,76 @@
+import { AbstractLevel } from 'abstract-level'
+import { Type } from '@sinclair/typebox'
+
+export interface Hook {
+  url: string
+  method: 'GET' | 'POST' | 'PUT' | 'DELETE'
+  headers: { [name: string]: string }
+}
+
+export const WebHookSchema = {
+  url: Type.String({
+    format: 'uri'
+  }),
+  method: Type.Union([Type.Literal('GET'), Type.Literal('POST'), Type.Literal('PUT'), Type.Literal('DELETE')]),
+  headers: Type.Record(Type.String(), Type.String())
+}
+
+export class HookStore {
+  db: AbstractLevel<any, string, any>
+
+  constructor (db: AbstractLevel<any, string, any>) {
+    this.db = db
+  }
+
+  async setHook (type: string, hook: Hook): Promise<void> {
```

Comment:
```suggestion
  async set (type: string, hook: Hook): Promise<void> {
```

what do you think of removing the word hook from the methods since it will likely be accessed as `hooks.set()` anyways